### PR TITLE
feat: release 0.2.8

### PR DESCRIPTION
**Pull request by igordanilcenko**
Fixed the exported activity bug when starting them with a wrong payload can cause a crash.

Additionally cleaned and improved the code base:
- Updated `README.md` and removed obsolete information (some parts of text marked as outdated)
- Updated Kotlin version to `1.9.22`
- Updated libs versions
- Increased lib and sample app versions to `0.2.8`
- Added the `.serializable()` extension for safely retrieving data from bundle
- Fixed some warnings
- Cleaned up code

### CODE REVIEW

File: yat-lib/src/main/java/yat/android/lib/YatIntegration.kt [hunk 1]
```diff
@@ -73,25 +74,43 @@ class YatIntegration {
     }
 
     sealed class Environment(val yatAPIBaseURL: String, val yatWebAppBaseURL: String) {
-        object SandBox : Environment("https://a.yat.fyi/", "https://yat.fyi/")
-
-        object Production : Environment("https://a.y.at/", "https://y.at/")
+        data object SandBox : Environment("https://a.yat.fyi/", "https://yat.fyi/")
```
Comment (TruszczynskiA) [inline yat-lib/src/main/java/yat/android/lib/YatIntegration.kt:77]: This environment is for internal use only. It should't be a part of the framework.

File: README.md [hunk 2]
```diff
@@ -103,75 +76,27 @@ class YatLib {
...
-                * YatLib.ColorMode.DARK for dark mode,
-                */
-                colorMode = YatLib.ColorMode.LIGHT,
-                /*
-                * Delegate interface is described below.
-                */
+                colorMode = YatIntegration.ColorMode.LIGHT, // Choose between LIGHT and DARK
                 delegate = this,
-                yatRecords = yatRecords
+                environment = YatIntegration.Environment.SandBox, // Choose between SandBox and Production
```
Comment (TruszczynskiA) [inline README.md:99]: User can configure the framework to connect with his own server. So there is more options than only sandbox and production.

File: README.md [hunk 3]
```diff
@@ -103,75 +76,27 @@ class YatLib {
...
+   
+            val config = YatConfiguration(
+                appReturnLink = "app://y.at?action", // deep link to return to the app
+                organizationName = "Yat Labs",
+                organizationKey = "yat",
             )
-            // initialize the library
-            YatLib.initialize(
+   
+            YatIntegration.setup(
```
Comment (TruszczynskiA) [inline README.md:94]: Why do you changed the naming convention from YatLib to YatIntegration?

**Review by TruszczynskiA — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (igordanilcenko) [reply README.md:94]: @TruszczynskiA No, the change `YatLib` -> `YatIntegration` was there since one of the previous releases

**Review by TruszczynskiA — APPROVED** [review]

**Pull request merged**